File: stripe/src/main/java/com/stripe/android/ApiKeyValidator.java
package com.stripe.android;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

final class ApiKeyValidator {
    private static final ApiKeyValidator DEFAULT = new ApiKeyValidator();

    @NonNull
    static ApiKeyValidator get() {
        return DEFAULT;
    }

    @NonNull
    String requireValid(@Nullable String apiKey) {
        if (apiKey == null || apiKey.trim().length() == 0) {
            throw new IllegalArgumentException("Invalid Publishable Key: " +
                    "You must use a valid publishable key to create a token. " +
                    "For more info, see https://stripe.com/docs/keys");
        }

        if (apiKey.startsWith("sk_")) {
            throw new IllegalArgumentException("Invalid Publishable Key: " +
                    "You are using a secret key, instead of the publishable one. " +
                    "For more info, see https://stripe.com/docs/keys");
        }

        return apiKey;
    }
}

File: stripe/src/main/java/com/stripe/android/ApiKeyValidator.kt
package com.stripe.android

internal class ApiKeyValidator {

    fun requireValid(apiKey: String?): String {
        require(!apiKey.isNullOrBlank()) {
            "Invalid Publishable Key: " +
                "You must use a valid Stripe API key to make a Stripe API request. " +
                "For more info, see https://stripe.com/docs/keys"
        }

        require(!apiKey.startsWith("sk_")) {
            "Invalid Publishable Key: " +
                "You are using a secret key instead of a publishable one. " +
                "For more info, see https://stripe.com/docs/keys"
        }

        return apiKey
    }

    internal companion object {
        private val DEFAULT = ApiKeyValidator()

        @JvmStatic
        internal fun get(): ApiKeyValidator {
            return DEFAULT
        }
    }
}
